api/krusty/remoteload_test.go

```diff
@@ -41,103 +28,254 @@ spec:
-	b := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
 	tmpDir, err := filesys.NewTmpConfirmedDir()
-	assert.NoError(t, err)
-	assert.NoError(t, fSys.WriteFile(filepath.Join(tmpDir.String(), "kustomization.yaml"), []byte(`
-resources:
-- github.com/kubernetes-sigs/kustomize/examples/multibases/dev/?ref=v1.0.6
-`)))
+	assert.NoError(err)
+	assert.NoError(fSys.WriteFile(filepath.Join(tmpDir.String(), "kustomization.yaml"), []byte(content)))
+	return fSys, tmpDir
+}
+
+func checkYaml(actual resmap.ResMap, expected string, assert *assert.Assertions) {
+	yml, err := actual.AsYaml()
+	assert.NoError(err)
+	assert.Equal(expected, string(yml))
+}
+
+func testRemoteResource(assert *assert.Assertions, kustomization string, expected string) {
+	fSys, tmpDir := createKusDir(kustomization, assert)
+
+	b := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
 	m, err := b.Run(
 		fSys,
 		tmpDir.String())
-	if utils.IsErrTimeout(err) {
-		// Don't fail on timeouts.
-		t.SkipNow()
+
+	assert.NoError(err)
+	checkYaml(m, expected, assert)
+
+	assert.NoError(fSys.RemoveAll(tmpDir.String()))
+}
+
+func TestRemoteLoad(t *testing.T) {
+	assert := assert.New(t)
+
+	fSys := filesys.MakeFsOnDisk()
+	b := krusty.MakeKustomizer(krusty.MakeDefaultOptions())
+
+	m, err := b.Run(
+		fSys,
+		"github.com/kubernetes-sigs/kustomize/examples/multibases/dev/?ref=v1.0.6")
+	assert.NoError(err)
+	checkYaml(m, multibaseDevExampleBuild, assert)
+}
+
+func TestRemoteResourceHttps(t *testing.T) {
+	tests := map[string]remoteResourceCase{
+		"basic": {
+			kustomization: `
+resources:
+- https://github.com/kubernetes-sigs/kustomize//examples/multibases/dev/?ref=v1.0.6`,
+		},
+		".git repo suffix, no slash suffix": {
+			kustomization: `
+resources:
+- https://github.com/kubernetes-sigs/kustomize.git//examples/multibases/dev?ref=v1.0.6`,
+		},
+		"repo": {
+			kustomization: `
+resources:
+- https://github.com/annasong20/kustomize-test.git?ref=main`,
+		},
+		"raw remote file": {
+			kustomization: `
+resources:
+- https://raw.githubusercontent.com/kubernetes-sigs/kustomize/v3.1.0/examples/multibases/base/pod.yaml
+namePrefix: dev-`,
+		},
+	}
+
+	for name, test := range tests {
+		test := test
+		if !test.skip {
+			t.Run(name, func(t *testing.T) {
+				testRemoteResource(assert.New(t), test.kustomization, multibaseDevExampleBuild)
+			})
+		}
 	}
-	assert.NoError(t, err)
-	yml, err := m.AsYaml()
-	assert.NoError(t, err)
-	assert.Equal(t, `apiVersion: v1
-kind: Pod
-metadata:
-  labels:
-    app: myapp
-  name: dev-myapp-pod
-spec:
-  containers:
-  - image: nginx:1.7.9
-    name: nginx
-`, string(yml))
-	assert.NoError(t, fSys.RemoveAll(tmpDir.String()))
 }
 
-func TestRemoteResourceGitHTTP(t *testing.T) {
-	output := `apiVersion: v1
-kind: Pod
-metadata:
-  labels:
-    app: myapp
-  name: dev-myapp-pod
-spec:
-  containers:
-  - image: nginx:1.7.9
-    name: nginx
-`
-	tests := []struct {
-		input []byte
-	}{
-		{
-			input: []byte(`
+func TestRemoteResourceSsh(t *testing.T) {
```

Comment:
Yes, the test passes locally. 

The test does throw an error. It panics on the next line. This happens because `assert` statements don't fail the test when the `assert`s fail. 

I do believe the test is failing because the server lacks SSH keys. I can set the `skip` field for the ssh tests to `true`. Is this good enough, or would you recommend creating the global variable?

Agreed, we can set up ssh keys in a different PR.

Reply:
> I do believe the test is failing because the server lacks SSH keys. I can set the skip field for the ssh tests to true.

Yup that's fine with me.